Source/Core/VideoCommon/VertexShaderGen.cpp

```diff
@@ -252,21 +252,22 @@ ShaderCode GenerateVertexShaderCode(APIType api_type, const ShaderHostConfig& ho
   out.Write("float4 pos = float4(dot(P0, rawpos), dot(P1, rawpos), dot(P2, rawpos), 1.0);\n");
   if ((uid_data->components & VB_HAS_NRM0) != 0)
   {
+    if ((uid_data->components & VB_HAS_NRM1) == 0)
+      out.Write("float3 rawnorm1 = " I_CACHED_BINORMAL ".xyz;\n");
+    if ((uid_data->components & VB_HAS_NRM2) == 0)
+      out.Write("float3 rawnorm2 = " I_CACHED_TANGENT ".xyz;\n");
+
     // Only the first normal gets normalized (TODO: why?)
```

Comment:
We should update that comment as we now know the answer.

The scale of the transform matrix is used to control the size of emboss map effect, by changing the scale of the transformed binormals (which only get used by emboss map)
By normalising the first transformed normal (which is used by lighting calculations and needs to be unit length), the same transform matrix can do double duty, scaling for emboss mapping, and not scaling for lighting.

Reply:
Done (also for `TransformUnit` and `UberShaderVertex`)